Src/AutoFixture/FreezeOnMatchCustomization.cs

```diff
@@ -43,8 +44,9 @@ public FreezeOnMatchCustomization(Type targetType)
         /// that will be satisfied by the frozen specimen.
         /// </param>
         /// <exception cref="ArgumentNullException">
-        /// <paramref name="targetType"/> or<paramref name="matcher"/> is null.
+        /// <paramref name="targetType"/> or <paramref name="matcher"/> is null.
         /// </exception>
+        [Obsolete("Please use FreezeOnMatchCustomization(object, IRequestSpecification) instead.")]
```

Comment:
Yes, it will be possible to get such exception with the [new constructor](https://github.com/sergeyshushlyapin/AutoFixture/blob/02a793050a14bc052c8320145ad9bc75280cac8f/Src/AutoFixture/FreezeOnMatchCustomization.cs#L80). That is the cost of allowing the constructor receive objects.

That is also the reason why I tried to avoid changing the public API of the `FreezeOnMatchCustomization` class introducing the [extended internal version of the customization](https://github.com/sergeyshushlyapin/AutoFixture/blob/4280168a380867bb288f5660849b1b22b491afaf/Src/AutoFixture.xUnit.net2/FreezeOnParameterMatchCustomization.cs). Unfortunately in this case we have to deal with both `TargetType` and `Request` properties [simultaneously](https://github.com/sergeyshushlyapin/AutoFixture/blob/4280168a380867bb288f5660849b1b22b491afaf/Src/AutoFixture/FreezeOnMatchCustomization.cs#L99). None of them will be obsoleted.